src/fs/file.rs

```diff
@@ -855,6 +855,17 @@ impl File {
     }
 }
 
+impl IntoRawFd for File {
+    // Consumes this object, returning the raw underlying file descriptor.
+    // Since all in-flight operations hold a reference to &self, the type-checker
+    // will ensure that there are no in-flight operations when this method is
+    // called.
```

Comment:
I don't understand this comment that says the in-flight operations hold a reference to &self. The in-flight operations I think you mean actually hold a cloned SharedFd, like in https://github.com/tokio-rs/tokio-uring/blob/afab7f2de5892759f816d03b3df4a20c1ac85fa9/src/io/write.rs#L10

Reply:
Ah yeah... when I was trying to wrap my head around SharedFd, I came to the following conclusion (please correct me If I am wrong): The SharedFd `close` method will hang forever if there is any in-flight IO. However, the actual File::close API does not appear to share this issue because the Rust type-checker already actively prevents in-flight asynchronous methods (or even just references to File) when transferring ownership of a tokio_uring::File (either by File::close or by File::into_raw_fd).

## Example 1: OK
```
fn main() -> std::io::Result<()> {
    tokio_uring::start(async {
        let file = File::open("test.txt").await?;
        let fd = file.into_raw_fd();
        let file = unsafe { File::from_raw_fd(fd) };
        file.close().await?;

        Ok(())
    })
}
```

## Example 2: Error
An async block that isn't awaited before calling `into_raw_fd` or `close` will result in an error.
```
fn main() -> std::io::Result<()> {
    tokio_uring::start(async {
        let file = File::open("test.txt").await?;

        tokio_uring::spawn(async {
            let buf: Vec<u8> = Vec::with_capacity(16);
            let (buf, res) = file.read_at(buf, 0).await;
        });

        let fd = file.into_raw_fd();
        let file = unsafe { File::from_raw_fd(fd) };
        file.close().await?;

        Ok(())
    })
}
```

## Example 3: Error
cannot move out of an `Rc`
```
fn main() -> std::io::Result<()> {
    tokio_uring::start(async {
        let file = std::rc::Rc::new(File::open("test.txt").await?);
        let file2 = file.clone();
        tokio_uring::spawn(async move {
            let buf: Vec<u8> = Vec::with_capacity(16);
            let (buf, res) = file2.read_at(buf, 0).await;
        });

        let fd = file.into_raw_fd();
        let file = unsafe { File::from_raw_fd(fd) };
        file.close().await?;

        Ok(())
    })
}
```

## Example 4: Ok
This compiles since we use try_unwrap (which ensures that there are no other references to File).
```
fn main() -> std::io::Result<()> {
    tokio_uring::start(async {
        let file = std::rc::Rc::new(File::open("test.txt").await?);
        let file2 = file.clone();
        tokio_uring::spawn(async move {
            let buf: Vec<u8> = Vec::with_capacity(16);
            let (buf, res) = file2.read_at(buf, 0).await;
        });

        match std::rc::Rc::try_unwrap(file) {
            Ok(file) => {
                let fd = file.into_raw_fd();
                let file = unsafe { File::from_raw_fd(fd) };
                file.close().await;
            },
            Err(err) => {
                panic!("unexpected io in-flight")
            }
        }

        Ok(())
    })
}
```